File: sem3/map/labs/a3-soniazagrean/src/domain/Product.java
import java.io.Serializable;

public class Product extends Entity implements Serializable {
    private String category;
    private String name;
    private int price;

    public Product(int id, String category, String name, int price) {
        super(id);
        this.category = category;
        this.name = name;
        this.price = price;
    }
    public Product() {}

    public String getCategory() {
        return this.category;
    }

    public String getName() {
        return this.name;
    }

    public int getPrice() {
        return this.price;
    }

    public void setCategory(String category) {
        this.category = category;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setPrice(int price) {
        this.price = price;
    }

    @Override
    public String toString() {
        return "Product:{" + getId() + "," + getCategory() + "," + getName() + "," + getPrice() + "}";
    }
}

File: sem2/oop/labs/Lab11/Product.cpp
//
// Created by Sonia Zagrean on 19.05.2025.
//

#include "Product.h"


Product::Product(int code, std::string name, int price) : code(code), name(name), price(price) {}

int Product::getCode() const {
    return code;
}

void Product::setCode(const int value) {
    code = value;
}

std::string Product::getName() const {
    return name;
}

void Product::setName(const std::string &value) {
    name = value;
}

int Product::getPrice() const {
    return price;
}

void Product::setPrice(const int value) {
    price = value;
}

bool Product::operator==(const Product &product) const {
    return (this->name == product.name && this->code == product.code && this->price == product.price);
}

Product& Product::operator=(const Product& other) {
    if (*this == other)
        return *this;
    name = other.name;
    code = other.code;
    price = other.price;
    return *this;
}
